src/node_order_optim/node_permute.h

```diff
@@ -18,7 +18,7 @@ void permute_nodeindices(Memb_list* ml, int* permute);
 void sort_ml(Memb_list* ml);
 
 // vec values >= 0 updated according to permutation
-void node_permute(int* vec, int n, int* permute);
+void node_permute(int* vec, int size, const std::vector<int>& permute);
```

Comment:
Maybe `permute_parent_index` would be even better?

Reply:
The problem with "permute" is that is sounds like one is moving a value to a new location. e.g. forward_permutation is
new[i] = old[perm[i]]
move the value at element index perm[i] to element index i.   Here we are merely (in place) updating the old parent[i]
(parent index of node index i) to the proper post-permutation parent node index
parent[i] = perm[parent[i]]
Maybe the name does not matter so much we always carry the following comment.
update_post_permute_parent_index(...); // parent[i] = perm[parent[i]]

Reply:
Since `update_parent_index(_v_parent_index)` is needed after `forward_permute(_v_parent_index)`, does it make sense to put these 2 in a single function?
For example:
```
void permute_parent_index(int* parent_index, int nodecount, const std::vector<int>& permute) {
    forward_permute(nt._v_parent_index, nt.end, p);
    update_parent_index(nt._v_parent_index, nt.end, perm);
}
```
instead of: https://github.com/neuronsimulator/nrn/blob/dccc4565e74e0819922e8478de94a375f78ea477/src/node_order_optim/cellorder.cpp#L424 and https://github.com/neuronsimulator/nrn/blob/dccc4565e74e0819922e8478de94a375f78ea477/src/node_order_optim/cellorder.cpp#L380?

Reply:
I think that's not that necessary since `update_parent_index` is used in multiple places